### PR TITLE
MARP-620 Add publish-release workflow

### PR DESCRIPTION
**Pull request by ndkhanh-axonivy**
This PR adds the Release Drafter workflow to the repository.

### CODE REVIEW

**Review by nqhoan-axonivy — CHANGES_REQUESTED** [review]
Hold on, please don't merge this PR:
the master file https://github.com/axonivy-market/github-workflows/blob/v4/.github/workflows/release-drafter.yml has no event workflow-call.
So @ndkhanh-axonivy please create a PR to adapt it, then maybe you must change this script to v5 or higher

File: .github/workflows/release-drafter.yml
Comment (ivy-rew) [inline .github/workflows/release-drafter.yml:17]: sorry, very late feedback but still:
- we are using maven here, and if we run a release it will not be revocable by normal means, so creating a 'prerelease' seems like confusing this reality. Do we have any real gain by this flag?
- version: the version is already state in the maven pom-files and automatically updated by the pipelines. So IMHO it makes no sense to let the user overwrite this version identifier. We should just stick to the maven version and also use it to tag and promote our releases.

finally the big benefit of it; we don't need any pipeline inputs and can design the release-drafter much simpler.

Comment (ndkhanh-axonivy) [reply .github/workflows/release-drafter.yml:17]: Hi @ivy-rew, thank you for your feedback. Here’s a summary of my thoughts and some clarifications:

1. **Supporting Release Notes for Older Tags:**
   - My goal is to allow users to create release notes for older tags. If the input version doesn’t match an existing tag, the system will create a new tag on the `HEAD` of the `master` branch and generate the release note based on that.

2. **Handling Pre-releases:**
   - Initially, I considered supporting release notes for pre-release versions (e.g., after a sprint in Scrum). However, after rethinking it, I realized the input isn’t crucial since we can manually mark a draft as a pre-release. I plan to remove this option, especially since it was originally intended for the project marketplace where we wanted sprint releases like `v1.0.0-m20` (milestone for sprint 20). However, this approach might not align well with other market items.

3. **Default Configuration in Release Drafter:**
   - The default configuration of Release Drafter uses the `$RESOLVED_VERSION` variable (see our [template](https://github.com/axonivy-market/.github/blob/master/.github/release-drafter.yml)), which follows [these rules](https://github.com/release-drafter/release-drafter?tab=readme-ov-file#version-resolver) to determine the version. It typically increments based on the latest version in the current release notes, starting at `v0.1.0` if none exist. If a PR labeled as `major` is included in the next release, the version will bump to `v1.0.0`.

### Summary
- By default, the version should be derived from the `pom.xml` not `$RESOLVED_VERSION`. Do you think it's a good idea to support generating release notes for older versions, or should this be handled as a custom workflow for specific market items?

Comment (ivy-rew) [reply .github/workflows/release-drafter.yml:17]: 1. Old releases
   - I wouldn't do it; every product is allowed to create releases back in time using the normal github release generator pages. They may not be as fancy as the release-drafter ones ... but will do the job to highlight changes by PRs ... and the user can manually adjust the content to whatever he likes.
2. Pre-Releases
   - I totally agree, a manual switch to 'pre-release' is sufficient: so we can drop this input.
3. Resolved versions
   - as far as I have seen the resolver suggests quit naturally the next best version, as soon as we have a first release tag. We need no special pom.xml handling, as the git-tags already correlate with the maven versions. So we are fine with the default and I think no input is required ... the release draft just should always be updated on merges to master. 

So I'd vote for not having any inputs on the release-drafter pipeline: keeping it lean and simple.
What we should consider though, is to introduce a new flag in the release.yml pipeline, that enforces the github release creation. This will publish the drafted release-notes and should be done by default. Though users should be able to turn it of with a flag. With this flag, maintainers without knowledge about releases and the changelog drafts, will automatically generate a nice changelog.

Comment (ndkhanh-axonivy) [reply .github/workflows/release-drafter.yml:17]: Hi @ivy-rew, I really appreciate your feedback and agree with points 1 and 2. However, I'm still a bit confused about the resolved versions, please correct me if I'm wrong.

1. After running the Release build, it generates PRs with two commits, one of which is tagged with the current version from the `pom.xml`. Based on this, I created a PR with a commit that had a tag (e.g. `v1.0.0`), and then merged it into `master` using the merge commit option.
   
2. My expectation was that after merging this PR, Release Drafter would generate the version based on the latest tag on `master`. However, the result was not what I expected - it generated the release note with version `v0.1.1` because the previous release note was at version `v0.1.0`.

Could you please help me check this? I've tried several times but still haven't succeeded. Or do you mean that we should manually edit the Release Drafter to select the correct version afterward?

Thanks in advance for your help!

Comment (ndkhanh-axonivy) [reply .github/workflows/release-drafter.yml:17]: And for the flag that allows users to publish the release instead of just creating a draft, please review this configuration to ensure it aligns with your expectations:

```yaml
name: Reusable Release Drafter

on:
  workflow_call:
    publish_release:
      required: false
      default: false
      type: boolean


permissions:
  contents: read

jobs:
  update_release_draft:
    permissions:
      contents: write
      pull-requests: write
    runs-on: ubuntu-latest
    steps:
      - uses: release-drafter/release-drafter@v6
        with:
          publish: ${{ github.event.inputs.publish_release }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

For any market item that wants to automatically publish the release, it would configure its pipeline like this:

```yaml
name: Release Drafter

on:
  push:
    branches:
      - master
  pull_request:
    types: [opened, reopened, synchronize]

permissions:
  contents: write
  pull-requests: write

jobs:
  build:
    uses: axonivy-market/github-workflows/.github/workflows/release-drafter.yml@v4
    with:
      publish_release: true
```

If this setup is correct, I do have a concern: it seems like this will create a release note every time the master branch is pushed, but we only want this to be triggered after the release build has been run. Could you help me address this issue? Your assistance would be a lifesaver.

Comment (ivy-rew) [reply .github/workflows/release-drafter.yml:17]: I think the automatic publishing should work with the solution described here: https://jacobtomlinson.dev/posts/2024/creating-github-releases-automatically-on-tags/
A second pipeline which runs `on: tag: v*.*.*` should only run, after you merge a release creation PR back to master. But I haven't tried this anywhere so you'd have to verify if that works for us.

release-publish


If such a release-publish.yml pipeline is not working as we think, I'd delay that automation and let the final release publishing remain a manual step that should be done by the releaser manually.

Comment (ndkhanh-axonivy) [reply .github/workflows/release-drafter.yml:17]: Hi @ivy-rew, thank you so much for your helpful suggestion! With your input, the release pipeline is now correctly obtaining the version after running.

However, I noticed that this approach triggers the publish release workflow immediately after a tag is published, rather than after merging a release PR into master. From my perspective, this is still acceptable because it aligns well with our process.

Please let me know if you have any further suggestions!

Comment (ivy-rew) [reply .github/workflows/release-drafter.yml:17]: Thanks a lot @ndkhanh-axonivy , really glad to see that these pipelines now are lean and powerful doing what we were looking for.

Great efforts you made here @ndkhanh-axonivy ; and I'm thankful for the professional and productive exchange we had here. It's an honor to collaborate with you :1st_place_medal:

Comment (ndkhanh-axonivy) [reply .github/workflows/release-drafter.yml:17]: Thank you so much, @ivy-rew ! I'm really happy to hear that you're pleased with the pipelines. It’s been a great experience working on this, and I appreciate your kind words and support.

I've learned a lot from this collaboration, and your professional and cooperative approach made everything smooth and productive. It’s truly an honor to work with you, and I look forward to more successful projects together! 🥇

**Review by ivy-rew — APPROVED** [review]
Perfect :100:

**Review by nqhoan-axonivy — APPROVED** [review]
LGTM

**Pull request merged**
